src/builds/mod.rs

```diff
@@ -0,0 +1,24 @@
+use error::{DashResult, DashError};
+
+pub mod buildbot;
+mod appveyor;
+mod travis;
+
+use url::{Url, Host};
+
+
+pub fn ingest_status_event(url: String) -> DashResult<()> {
+    if let Ok(url) = Url::parse(&url) {
+        if let Some(segments) = url.path_segments() {
+            let build = segments.last().unwrap();
```

Comment:
So according to the [docs](https://docs.rs/url/1.4.1/url/struct.Url.html#method.path_segments), the `segments` iterator should always contain at least one string, so AFAICT the `unwrap()` should never panic.

I'll go ahead and change the `unwrap` to an `expect` though.